### ISSUE TITLE
Selenide missed custom user-agent part after comma in Google Chrome browser

### ISSUE THREAD

**Issue by FrameBassman** [selenide/selenide#697]
## The problem
`ChromeDriverFactory.transferChromeOptionsFromSystemProperties` method splits args wrong.

## Details
I want to open page with `A,B` user-agent in Google Chrome browser.
I set `chromeoptions.args` environment variable like this:

```
String userAgentValue = "A,B";
System.setProperty("chromeoptions.args", "--user-agent=" + userAgentValue);
```
Then browser opens page with `user-agent=A` headers. Part after comma was missed.

## Tell us about your environment
* **Selenide Version** 4.10.01
* **Chrome\Firefox\IE Version:** Google Chrome 63.0.3239.108
* **Browser Driver Version:** ChromeDriver 63.0.3239.108
* **Selenium Version:** 3.8.1
* **OS Version:** macOS High Sierra 10.13.3

## Code To Reproduce Issue
```
    @Test
    public void userAgentWithCommas() {
        Configuration.browser = "chrome";
        String userAgentValue = "A,B";
        System.setProperty("chromeoptions.args", "--user-agent=" + userAgentValue);
        open("https://google.com");
        open("https://google.com");
    }
```

**Comment by vinogradoff** [comment]
at the moment comma is a separator between different options like in

`--user-agent=abc,--maximize-window=true` etc.

that is the reason why it is ignored.

Unfortunately User-Agent could use  almost every other separator as well like in

Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36

We can only use ASCII symbols because we may want to use it from command line like in

`-Dchromeoption.args=....`

So, we probably should handle "" for this cases for your example

--user-agent="A,B",anotherOption=....

Selenide should drop the "" when giving the params over to WebDriver. Not very complicated but not a one-liner :)

Can you make a PR for us with the solution? :)

**Comment by FrameBassman** [comment]
You can assign this issue to me and I fix it ASAP)

**Comment by vinogradoff** [comment]
Your example to test with would be 

`System.setProperty("chromeoptions.args", "--user-agent=" +"\"" + userAgentValue + "\"");`

**Comment by fyrewall77** [comment]
I think the fix is probably something along the lines of:

ChromeDriverFactory.java:
ChromeOptions createChromeOptions(Proxy proxy) {
...
     List<String> options = splitOnCommaAndNotOnQuotes(chromeSwitches);
      options.addArguments(options);
...
   List<String> splitOnCommaAndNotOnQuotes(String s) {
    return Splitter.on(Pattern.compile(",(?=(?:[^\"]*\"[^\"]*\")*[^\"]*$)")).split(s);
   }

**Comment by FrameBassman** [comment]
@vinogradoff Could you provide me permissions for code committing?

**Comment by vinogradoff** [comment]
You should work in a branch, everybody can commit in a branch. The it will
be review and merged

Viele Grüße

Alexei Vinogradov

On 26. Februar 2018 at 21:37:43, Dmitry Romashov (notifications@github.com)
wrote:

@vinogradoff <https://github.com/vinogradoff> Could you provide me
permissions for code committing?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/codeborne/selenide/issues/697#issuecomment-368641574>,
or mute the thread

**Comment by fyrewall77** [comment]
You have my permission...

**Comment by vinogradoff** [comment]
@fyrewall77 @FrameBassman ideally add a util method e.g. "splitStringTakingCareOfQuotes(Char/String separator)" (separator=',')

and write enough unit tests for it like
abc,"ab,bc",abc ->abc ab,bc abc 
abc,"ab\",bc",abc -> abc ab",bc abc
abc,\"ab,bc\",abc -> abc "ab bc" abc

**Comment by FrameBassman** [comment]
I suggest to escape all important commas in arguments via backslash (\) character

**Comment by VatslauX** [comment]
Is the same for FF or Chrome only?

**Comment by symonk** [comment]
@rosolko @asolntsev @vinogradoff is this solved with #883 ? #836 got closed

**Comment by vinogradoff** [comment]
Yes, solved by #883
